src/core/server/http/logging/get_payload_size.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import type { ReadStream } from 'fs';
+import { isBoom } from '@hapi/boom';
+import type { Request } from '@hapi/hapi';
+
+/**
+ * Attempts to determine the size (in bytes) of a Hapi response
+ * body based on the payload type. Falls back to `undefined`
+ * if the size cannot be determined from the response object.
+ *
+ * @param response Hapi response object or Boom error
+ *
+ * @internal
+ */
+export function getResponsePayloadBytes(response: Request['response']): number | undefined {
```

Comment:
The way we did this in legacy was wildly inaccurate (even if https://github.com/elastic/kibana/issues/80544 was not an issue), because we were assuming the the correct content length could always be calculated by stringifying the response payload.

What follows is my attempt at a more accurate mechanism for extracting payload size from hapi responses, taking into account differences between Boom errors, buffers, streams, objects, and plain text responses, and falling back to the `content-length` header when available.

Since I don't think any of this information is mission-critical and worth crashing Kibana over, I'm purposely swallowing errors here, returning only `undefined` if the size can't be determined for any reason.